weather.py

```diff
@@ -2,116 +2,106 @@
 import sys
 import csv
 from datetime import datetime
+from pathlib import Path
 
-
-months =[]
+data_folder = Path("C:/Som/Personal Projects/weatherdata/")
 
 def main():
     if str.upper(sys.argv[1]) == 'YEAR':
         yr = sys.argv[2]
-        val1 = yearcalc(yr)
-        reportone(val1)
+        yr_data = year_calculator(yr)
+        yearly_report(yr_data)
     elif str.upper(sys.argv[1]) == 'MONTH':
-        mn = sys.argv[2].split('/')
-        val2 = monthcalc(mn)
-        reporttwo(val2)
+        mo = sys.argv[2].split('/')
+        mo_data = month_calculator(mo)
+        monthly_report(mo_data)
     else:
         print("Please check input parameters")
         exit
 
 
-def yearcalc(y):
-    mx_tmp = 0
-    mx_hum = 0
-    mn_tmp = 100
-    os.chdir(r'C:\Som\Personal Projects\weatherdata')
-    for f in os.listdir():
-        if f.__contains__(y):
-            months.append(f)
-    for month in months:
-        with open (f'C:\\Som\\Personal Projects\\weatherdata\\{month}','r') as data_file:
-            #with open (r'C:\Som\Personal Projects\weatherdata\lahore_weather_1996_Dec.txt','r') as data_file:
-            csv_data = csv.reader(data_file, delimiter=',')
-            csv_data.__next__()
-            csv_data.__next__()
-            for line in csv_data:
-                if line[0].__contains__('<!'):
-                    break
-                else:
-                    if line[1] == '' and line[3] == '' and line[7] == '':
-                        continue
+def year_calculator(yr):
+    max_tmp = 0
+    max_hum = 0
+    min_tmp = 100
```

Comment:
May be you can also update these `max_temperature`, `max_humidity`, `min_temperature`.

Reply:
And also, `yr` to `year`

Reply:
ok